python/src/aac/lang/definition_helpers.py

```diff
@@ -0,0 +1,233 @@
+            else:
+                logging.error("keys exceeds depth of the dict to search")
+                done = True
+        done = True
+
+    # not an error, just zero results
+    else:
+        logging.info(f"keys[{search_keys}] not found in model")
+        done = True
+
+    return ret_val
+
+
+def search_definition(parsed_definition: ParsedDefinition, search_keys: list[str]) -> list:
+    """
+    Search an AaC definition structure by key(s).
+
+    Searches a ParsedDefinition for a subset of the definition contents given a set of keys.
+    Search returns a list of  entries in the definition that correspond to those keys. This search
+    will traverse the entire parsed definition's structure.
+
+        Typical usage example:
+        Let's say you have a definition structure parsed from the following AaC yaml.
+
+            data:
+            name: MyData
+            fields:
+                - name: one
+                type: string
+                - name: two
+                type: string
+            required:
+                - one
+
+        You know the structure of the specification and need to iterate through each field.
+        The search utility method simplifies that for you.
+
+            for field in definition_helpers.search(my_model, ["data", "fields"]):
+                print(f"field_name: {field["name"]} field_type {field["type"]}")
+
+        The above example demonstrates a complex type being returned as a dict.  Search will also
+        provide direct access to simple types in the model.
+
+            for field_name in definition_helpers.search(my_definition, ["data", "fields", "name"]):
+                print(f"field_name: {field_name}")
+
+    Args:
+        model: The definition to search.  This is often the value taken from the dict returned
+            by aac.parser.parse(<aac_file>).
+        search_keys: A list of strings representing keys in the model dict hierarchy.
+
+    Returns:
+        Returns a list of found data items based on the search keys.
+
+    """
+
+    return search(parsed_definition.definition, search_keys)
+
+
+def get_models_by_type(models: dict[str, dict], root_name: str) -> dict[str, dict]:
+    """Gets subset of models of a specific root name.
+
+    The aac.parser.parse() returns a dict of all parsed types.  Sometimes it is
+    useful to only work with certain roots (i.e. model or data).  This utility
+    method allows a setup of parsed models to be "filtered" to a specific root name.
+
+    Args:
+        models: The dict of models to search.
+        root_name: The root name to filter on.
+
+    Returns:
+        A dict mapping type names to AaC model definitions.
+    """
+    ret_val = {}
+    for key, value in models.items():
+        if root_name in value.keys():
+            ret_val[key] = value
+
+    return ret_val
+
+
+def get_definitions_by_root_key(parsed_definitions: list[ParsedDefinition], root_key: str) -> list[ParsedDefinition]:
```

Comment:
For these definition helpers, the args should look like:
```suggestion
def get_definitions_by_root_key(root_key: str, parsed_definitions: list[ParsedDefinition]) -> list[ParsedDefinition]:
```